src/lib.rs

```diff
@@ -86,53 +88,36 @@ pub fn db_conn_string(config: &Config) -> Result<String> {
         .collect::<Result<_, _>>()
         .context(ErrorKind::ConfigMissingKeys)?;
 
-    match string_vec[0].as_str()
-        {
-            "postgres" | "postgresql" => (),
-            _ => Err(ErrorKind::UnsupportedDbScheme)?,
-        }
-
-    Ok(format!(
-        "{scheme}://{username}:{password}@{host}:{port}/{database}",
-        scheme = string_vec[0],
-        username = string_vec[1],
-        password = string_vec[2],
-        host = string_vec[3],
-        port = string_vec[4],
-        database = string_vec[5],
-    ))
-}
+    let supported_schemes = ["postgres", "postgresql", "mysql", "sqlite"];
+    if !supported_schemes.contains(&string_vec[0].as_str()) {
+        Err(ErrorKind::UnsupportedDbScheme)?;
+    }
 
-#[derive(Debug, thiserror::Error)]
-#[error("Configuration was missing expected keys: [{:?}]", _0)]
-pub struct MissingKeys(Vec<String>);
-
-#[derive(Clone, Copy, Debug, Eq, thiserror::Error, Hash, PartialEq)]
-pub enum ErrorKind {
-    #[error("Unsupported database scheme, only 'postgres' and 'postgresql' are allowed.")]
-    UnsupportedDbScheme,
-    #[error("Configuration was missing expected keys")]
-    ConfigMissingKeys,
-    #[error("Config struct cannot be modified")]
-    ConfigImmutable,
+    Ok(concat!(
+        string_vec[0], "://", string_vec[1], ":", string_vec[2], "@", string_vec[3], ":", string_vec[4], "/", string_vec[5],
+    ).to_string())
```

Comment:
**Suggestion:** In the `db_conn_string` function, you are concatenating strings using `concat!` macro. This can be inefficient for large number of strings. Consider using `format!` macro instead.
```suggestion
    Ok(format!("{}://{}:{}@{}:{}/{}", string_vec[0], string_vec[1], string_vec[2], string_vec[3], string_vec[4], string_vec[5]))
```